include/torch-mlir-dialects/Dialect/TMTensor/IR/TMTensorOps.td

```diff
@@ -314,7 +314,7 @@ def TMTensor_AttentionOp : TMTensor_Op<"attention",
       return getOutputType().getRank();
     }
     int64_t getIterationDomainRank() {
-      return 2;
+      return getKeyRank() - 2;
```

Comment:
I ripped out the getIterationDomainRank. It turns out its completely irrelevant. Because of the multi-loop behavior of flash attention we actually have multiple adjacent loops with different iteration domains, so this is effectively stubbed out.

I was wondering why it didn't matter what this value was, tests always passed.